### PR TITLE
 18 key bluetooth NUMBERPAD

### PR DESCRIPTION
**Pull request by latincompass**
## Description

 18 bluetoothpad  , use the AdafruitBLE SPI FRIEND

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [x] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by latincompass** [comment]
Sir , I can`t find the modify by myself a hole day.so i closed #9165,open a new PR. SORRY

File: keyboards/18bluetoothpad/config.h [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.*/
+
+#pragma once
+
+#include "config_common.h"
+#define VENDOR_ID       0x6C63 // "lc" = latincompass
+#define PRODUCT_ID      0x6C70 // "1p" = 18bluetoothpad
```
Comment (fauxpark) [inline keyboards/18bluetoothpad/config.h:18]:
```suggestion
#define PRODUCT_ID      0x3170 // "1p" = 18bluetoothpad
```
ASCII `1` = 0x31

File: keyboards/18bluetoothpad/config.h [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+#define DESCRIPTION     18 RGB bluetoothpad
+
+/* key matrix size */
+#define MATRIX_ROWS 5
+#define MATRIX_COLS 4
+
+
+ #define MATRIX_ROW_PINS { B4, B5, B6, B7, C7 }
+
+ #define MATRIX_COL_PINS { D4, D5, D6, C6 }
```
Comment (fauxpark) [inline keyboards/18bluetoothpad/config.h:31]:
```suggestion
#define MATRIX_ROW_PINS { B4, B5, B6, B7, C7 }
#define MATRIX_COL_PINS { D4, D5, D6, C6 }
```

**Comment by latincompass** [comment]
I try a lot of times. I can`t   rollback the  #9165 Program PR. SORRY

File: keyboards/18bluetoothpad/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        KC_P4, KC_P5, KC_P6, KC_KP_PLUS,
+        KC_P1, KC_P2, KC_P3, KC_KP_MINUS,
+        KC_P0, KC_PDOT,KC_DELETE, KC_KP_ENTER,
+        KC_PGUP, KC_PGDOWN),
+    [1] = LAYOUT_all(
+        RGB_TOG, RGB_MOD, RGB_M_K, RGB_M_X,
+        RGB_SAI, RGB_SAD, RGB_HUI, RGB_HUD,
+        RGB_VAI, RGB_VAD, RGB_SPI, RGB_SPD,
+        RGB_SPI, RGB_SPD, RGB_TOG, KC_TRNS,
+        KC_NUMLOCK, KC_CALCULATOR),
```
Comment (fauxpark) [inline keyboards/18bluetoothpad/keymaps/via/keymap.c:41]: You need two more layers here (filled with `KC_TRNS`) for the default `DYNAMIC_KEYMAP_LAYER_COUNT` of 4.

File: keyboards/18bluetoothpad/readme.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
+18bluetoothpad
```
Comment (fauxpark) [inline keyboards/18bluetoothpad/readme.md:1]:
```suggestion
# 18bluetoothpad
```

File: keyboards/18bluetoothpad/readme.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+![18bluetoothpad](https://github.com/latincompass/18bluetoothpad/blob/master/ZZ20200421113009.png)
+![18bluetoothpad](https://github.com/latincompass/18bluetoothpad/blob/master/IMG_0116.JPG)
+![18bluetoothpad](https://github.com/latincompass/18bluetoothpad/blob/master/IMG_0117.JPG)
```
Comment (fauxpark) [inline keyboards/18bluetoothpad/readme.md:5]:
```suggestion
```

File: keyboards/18bluetoothpad/readme.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+18 RGB bluetoothpad(Adafruit SPI FRIEND)
+
+Keyboard Maintainer: latincompass
+Hardware Availability: https://github.com/latincompass/18bluetoothpad
+Make example for this keyboard (after setting up your build environment):
+
+make 18bluetoothpad:default
+See the build environment setup and the make instructions for more information. Brand new to QMK? Start with our Complete Newbs Guide.
```
Comment (fauxpark) [inline keyboards/18bluetoothpad/readme.md:14]:
```suggestion
* Keyboard Maintainer: latincompass
* Hardware Availability: https://github.com/latincompass/18bluetoothpad

Make example for this keyboard (after setting up your build environment):

    make 18bluetoothpad:

See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```

File: keyboards/18bluetoothpad/rules.mk [hunk 7]
```diff
@@ -0,0 +1,28 @@
+# MCU name
```
Comment (fauxpark) [inline keyboards/18bluetoothpad/rules.mk:1]: Please change this file's line endings to Unix.

File: keyboards/18bluetoothpad/readme.md [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+![18bluetoothpad](https://github.com/latincompass/18bluetoothpad/blob/master/ZZ20200421113009.png)
+
+18 RGB bluetoothpad(Adafruit SPI FRIEND)
+
+* Keyboard Maintainer: latincompass
+* Hardware Availability: https://github.com/latincompass/18bluetoothpad
+
+Make example for this keyboard (after setting up your build environment):
+
+    make 18bluetoothpad:
```
Comment (fauxpark) [inline keyboards/18bluetoothpad/readme.md:12]:
```suggestion
    make 18bluetoothpad:default
```
oops

File: keyboards/18bluetoothpad/keymaps/via/keymap.c (hunk 3 above)
Comment (latincompass) [reply keyboards/18bluetoothpad/keymaps/via/keymap.c:41]: immediately，sir

File: keyboards/18bluetoothpad/rules.mk (hunk 7 above)
Comment (latincompass) [reply keyboards/18bluetoothpad/rules.mk:1]: Sir , i don`t understand you talk,sorry

File: keyboards/18bluetoothpad/rules.mk [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+NKRO_ENABLE = no            # USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+AUDIO_ENABLE = no
+RGBLIGHT_ENABLE = yes
+OLED_DRIVER_ENABLE = no
+ENCODER_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/18bluetoothpad/rules.mk:28]:
```suggestion
# MCU name
MCU = atmega32u4

# Processor frequency
F_CPU = 8000000

# Bootloader selection
#   Teensy       halfkay
#   Pro Micro    caterina
#   Atmel DFU    atmel-dfu
#   LUFA DFU     lufa-dfu
#   QMK DFU      qmk-dfu
#   ATmega32A    bootloadHID
#   ATmega328P   USBasp
BOOTLOADER = atmel-dfu

# Build Options
# change yes to no to disable
#
BOOTMAGIC_ENABLE = yes      # Virtual DIP switch configuration
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
MIDI_ENABLE = no            # MIDI support
BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
AUDIO_ENABLE = no           # Audio output
FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
BLUETOOTH = AdafruitBLE
```

File: keyboards/18bluetoothpad/config.h (hunk 1 above)
Comment (latincompass) [reply keyboards/18bluetoothpad/config.h]: ok

File: keyboards/18bluetoothpad/rules.mk (hunk 9 above)
Comment (latincompass) [reply keyboards/18bluetoothpad/rules.mk]: Approved

File: keyboards/18bluetoothpad/config.h (hunk 2 above)
Comment (latincompass) [reply keyboards/18bluetoothpad/config.h]: Approved

File: keyboards/18bluetoothpad/readme.md (hunk 6 above)
Comment (latincompass) [reply keyboards/18bluetoothpad/readme.md]: Approved

File: keyboards/18bluetoothpad/config.h (hunk 1 above)
Comment (latincompass) [reply keyboards/18bluetoothpad/config.h]: Approved

File: keyboards/18bluetoothpad/readme.md (hunk 5 above)
Comment (latincompass) [reply keyboards/18bluetoothpad/readme.md]: Approved

File: keyboards/18bluetoothpad/readme.md (hunk 4 above)
Comment (latincompass) [reply keyboards/18bluetoothpad/readme.md]: Approved

**Pull request closed without merging**